content/blog/2020-01-vroom-1-1-0/index.Rmarkdown

```diff
@@ -0,0 +1,131 @@
+But for many datasets `vroom::vroom()` can read them much, much faster (hence the name).
+Get the latest version with:
+
+```{r, eval = FALSE}
+install.packages("vroom")
+```
+
+And attach the package by running
+
+```{r}
+library(vroom)
+```
+
+Improvements in this release include a hex logo, support for big integer data, improved delimiter guessing, including delimiters in specifications, and streamlined reading from standard input.
+
+See the [change log](https://vroom.r-lib.org/news/index.html) for a full list of changes and bug fixes in this version.
+
+## Hex logo
+
+Thanks to [Allison Horst](https://github.com/allisonhorst/stats-illustrations) we now have an awesome hex logo for vroom!
+
+![](logo.png)
+
+## Big integer support
+
+R's standard integers are stored in 32 bits of binary data, which means that the largest value they can store is 2,147,483,647 (`2^31 - 1`).
+R implicitly converts integers for most operations with doubles to 64 bit floating point values, which is why you may not have noticed this limitation before.
+
+```{r}
+options(digits = 22)
+x <- 2147483647L
+str(x)
+str(x + 1L)
+str(x + 1)
+x + 1
+```
+
+However even 64 bit floating point values can only store consecutive integers up to 9,007,199,254,740,992 (`2^53`) without losing precision.
+You can observe this because if you try adding 1 to this number you will get the same number.
+
+```{r}
+y <- 9007199254740992
+z <- y + 1
+z
+y == z
+```
+
+To store consecutive integers bigger than this you need to use a 64 bit integer type.
+R does not have native support for 64 bit integers, however the [bit64](https://cran.r-project.org/package=bit64) package provides support for them.
+Because these integers are so large they rarely occur in real world data, however they can often be obtained from generated data, such as database identifiers.
+
+vroom 1.1.0 now supports reading these big integers into the `integer64` type provided by bit64 with a new `col_big_integer()` column type (shortcut 'I').
+
+```{r}
+x <- vroom("id\n9007199254740993\n", col_type = "I", delim = ",")
+x
+
+x$id + 1
+```
+
+## Improved delimiter guessing
+
+The code to guess delimiters has been rewritten which should make it more robust to most inputs.
+Previous versions of vroom would fall back to using a newline delimiter if a delimiter could not be guessed.
+vroom 1.1.0 instead throws an error.
+
+```{r, error = TRUE}
+vroom("x\n1\n")
+vroom("x\n1\n", delim = ",")
+```
+
+## Delimiters in the specification
+
+vroom now includes the delimiter in the specification object, which means you no longer have to separately provide the delimiter if you are using an existing specification.
+
+```{r}
+# read a csv file, the delimiter is provided as ','
+x <- vroom(vroom_example("mtcars.csv"), delim = ',')
+mtcars_spec <- spec(x)
+
+# If the file is read again with the spec, no need to provide the delimiter
+y <- vroom(vroom_example("mtcars.csv"), col_types = mtcars_spec)
+```
```

Comment:
Non-editorial note: This is great, and if we don't have it somewhere else, we should!